spec/jobs/backorder_job_spec.rb

```diff
@@ -105,6 +109,27 @@
       perform_enqueued_jobs(only: CompleteBackorderJob)
     end
 
+    it "skips unavailable items" do
+      order.order_cycle = create(
+        :simple_order_cycle,
+        distributors: [order.distributor],
+        variants: [beans],
+      )
+      beans.on_demand = true
+      beans.on_hand = -3
+      beans.semantic_links << SemanticLink.new(
+        semantic_id: "#{product_link}-removed"
+      )
+
+      stub_request(:get, catalog_url).to_return(body: catalog_json)
+
+      expect {
+        subject.place_backorder(order)
+      }.not_to change { beans.on_hand }
+
+      # And no error was raised, no order was placed.
+    end
```

Comment:
This only testing a product removed from the catalog, shouldn't we also test the empty backorder and the zero quantity scenario ?

Reply:
I can imagine the shop manager would like to know that the backorder failed.
If they are listing a product for sale, that they can no longer order from supplier, then they should probably remove the product from their shopfront or take some other action.
So would it be worth sending an email to the shop managers here?

Reply:
Maybe. The latest discussion with Garethe revealed that local stock can still be sold. So a new case to consider in my other PR:

* Product is on-demand and imported to DFC.
* Product is mapped to wholesale variant.
* At the end of one cycle, OFN still has 5 retail items left in stock.
* At the next cycle, the product disappears from the catalog.
* The product needs to switch to stock-controlled and keep the 5 in stock.
* The product can still be sold until sold out.
* We don't need to send an error message in this case. At least it hasn't been requested.
* The item may come back into the catalog, in which case we can resume ordering.

Reply:
> This only testing a product removed from the catalog, shouldn't we also test the empty backorder and the zero quantity scenario ?

Empty backorder: this is tested here as well. No backorder is placed, otherwise there would be an HTTP request to place an order which would fail as we didn't stub anything.

Zero quantity: This is covered as well. When the catalog is missing the product, we return the zero quantity and then we skip that product.

Reply:
thanks for the clarification !